test/framework/src/main/java/org/elasticsearch/repositories/blobstore/BlobStoreTestUtil.java

```diff
@@ -118,6 +123,27 @@ private static void assertIndexGenerations(BlobContainer repoRoot, long latestGe
         assertTrue(indexGenerations.length <= 2);
     }
 
+    private static void assertShardIndexGenerations(BlobContainer repoRoot, ShardGenerations shardGenerations) throws IOException {
+        final BlobContainer indicesContainer = repoRoot.children().get("indices");
+        for (IndexId index : shardGenerations.indices()) {
+            final List<String> gens = shardGenerations.getGens(index);
+            if (gens.isEmpty() == false) {
+                final BlobContainer indexContainer = indicesContainer.children().get(index.getId());
+                final Map<String, BlobContainer> shardContainers = indexContainer.children();
+                for (int i = 0; i < gens.size(); i++) {
+                    final String generation = gens.get(i);
+                    assertThat(generation, not(ShardGenerations.DELETED_SHARD_GEN));
+                    if (generation != null && generation.equals(ShardGenerations.NEW_SHARD_GEN) == false) {
```

Comment:
assertConsistency is for what's on disk? The repositoryData should not contain `NEW_SHARD_GEN` entries?

Reply:
> assertConsistency is for what's on disk?

Yes

> The repositoryData should not contain NEW_SHARD_GEN entries?

See above, I think it should contain `NEW_SHARD_GEN` entries.